packages/babel-parser/src/plugins/typescript.js

```diff
@@ -659,6 +659,33 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return this.finishNode(node, "TSLiteralType");
     }
 
+    tsParseTemplateLiteralType(): N.TsType {
+      const node: N.TsLiteralType = this.startNode();
+      const startPosition = this.state.start;
+      const startLocation = this.state.startLoc;
+      const templateNode = this.parseTemplate(false);
+      if (
+        templateNode.expressions.length === 0 &&
+        templateNode.quasis.length === 1
+      ) {
+        const value = templateNode.quasis[0].value.raw;
+        const noSubstitutionTemplateLiteralNode = this.startNodeAt(
+          startPosition,
+          startLocation,
+        );
+        this.addExtra(node, "rawValue", value);
+        this.addExtra(node, "raw", value);
+        node.value = value;
+        node.literal = this.finishNode(
+          noSubstitutionTemplateLiteralNode,
+          "NoSubstitutionTemplateLiteral",
+        );
```

Comment:
I don't see why we need a new node type here. We can just re-use `TemplateLiteral`: if `node.expressions.length === 0`, then it doesn't have any substitution.